packages/agent-toolkit/src/shared/parameters.ts

```diff
@@ -9,123 +11,127 @@ const addressSchema = z.string().regex(/^0x[a-fA-F0-9]{40}$/);
-    )
-    .optional()
-    .describe("The metadata to store with the bucket"),
-});
+export const createBucketParameters = (_context: Context = {}) =>
+  z.object({
+    bucketAlias: z
+      .string()
+      .min(1)
+      .describe("The alias of the bucket to create"),
+    metadata: z
+      .record(
+        z.string().min(1).describe("The key of the metadata"),
+        z.string().min(1).describe("The value of the metadata"),
+      )
+      .optional()
+      .describe("The metadata to store with the bucket"),
+  });
 
 /**
  * Parameters for the getOrCreateBucket function
  * @param bucketAlias - The alias of the bucket to retrieve or create
  * @param metadata - The metadata to store with the bucket (optional)
  */
-export const getOrCreateBucketParameters = z.object({
-  bucketAlias: z
-    .string()
-    .describe("The alias of the bucket to retrieve or create"),
-  metadata: z
-    .record(
-      z.string().min(1).describe("The key of the metadata"),
-      z.string().min(1).describe("The value of the metadata"),
-    )
-    .optional()
-    .describe("The metadata to store with the bucket"),
-});
+export const getOrCreateBucketParameters = (_context: Context = {}) =>
+  z.object({
+    bucketAlias: z
+      .string()
+      .describe("The alias of the bucket to retrieve or create"),
+    metadata: z
+      .record(
+        z.string().min(1).describe("The key of the metadata"),
+        z.string().min(1).describe("The value of the metadata"),
+      )
+      .optional()
+      .describe("The metadata to store with the bucket"),
+  });
 
 /**
  * Parameters for the addObject function
  * @param bucket - The address of the bucket
  * @param key - The key under which to store the object
- * @param data - The data to store as a string, File, or Uint8Array
+ * @param data - The data to store as a string value
  * @param metadata - The metadata to store with the object
  * @param overwrite - Whether to overwrite existing data at that key
  */
-export const addObjectParameters = z.object({
-  bucket: addressSchema.describe("The address of the bucket"),
-  key: z.string().min(1).describe("The key under which to store the object"),
-  data: z
-    .union([z.string(), z.instanceof(File), z.instanceof(Uint8Array)])
-    .describe("The data to store as a string, File, or Uint8Array value"),
-  metadata: z
-    .record(
-      z.string().min(1).describe("The key of the metadata"),
-      z.string().min(1).describe("The value of the metadata"),
-    )
-    .optional()
-    .describe("The metadata to store with the object"),
-  overwrite: z
-    .boolean()
-    .optional()
-    .describe("Whether to overwrite existing data at that key"),
-});
+export const addObjectParameters = (_context: Context = {}) =>
+  z.object({
+    bucket: addressSchema.describe("The address of the bucket"),
+    key: z.string().min(1).describe("The key under which to store the object"),
+    data: z.string().describe("The data to store as a string value"),
```

Comment:
now, we just assume that when you add an object, it's a string (instead of allowing uint8arrays or files, which confused the model)